### ISSUE TITLE
Build errors with Clang: error: unknown type name 'u_int', 'u_char', 'u_short'

### ISSUE THREAD

**Issue by barracuda156** [chrberger/stringtoolbox#3]
```
In file included from /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_stringtoolbox/stringtoolbox/work/stringtoolbox-0.0.4/test/Test-stringtoolbox.cpp:26:
  In file included from /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_stringtoolbox/stringtoolbox/work/stringtoolbox-0.0.4/test/catch.hpp:6956:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/sysctl.h:83:
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/ucred.h:101:2: error: unknown type name 'u_int'
          u_int   cr_version;             /* structure layout version */
          ^
  In file included from /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_stringtoolbox/stringtoolbox/work/stringtoolbox-0.0.4/test/Test-stringtoolbox.cpp:26:
  In file included from /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_stringtoolbox/stringtoolbox/work/stringtoolbox-0.0.4/test/catch.hpp:6956:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/sysctl.h:85:
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/proc.h:116:2: error: unknown type name 'u_int'
          u_int   p_estcpu;        /* Time averaged value of p_cpticks. */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/proc.h:121:2: error: unknown type name 'u_int'
          u_int   p_swtime;        /* Time swapped in or out. */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/proc.h:122:2: error: unknown type name 'u_int'
          u_int   p_slptime;       /* Time since last blocked. */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/proc.h:136:2: error: unknown type name 'u_char'; did you mean 'char'?
          u_char  p_priority;     /* Process priority. */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/proc.h:137:2: error: unknown type name 'u_char'; did you mean 'char'?
          u_char  p_usrpri;       /* User-priority based on p_cpu and p_nice. */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/proc.h:142:2: error: unknown type name 'u_short'; did you mean 'short'?
          u_short p_xstat;        /* Exit status for wait; also stop signal. */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/proc.h:143:2: error: unknown type name 'u_short'; did you mean 'short'?
          u_short p_acflag;       /* Accounting flags. */
          ^
  In file included from /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_stringtoolbox/stringtoolbox/work/stringtoolbox-0.0.4/test/Test-stringtoolbox.cpp:26:
  In file included from /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_stringtoolbox/stringtoolbox/work/stringtoolbox-0.0.4/test/catch.hpp:6956:
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/sys/sysctl.h:770:23: error: unknown type name 'u_int'
  int     sysctl(int *, u_int, void *, size_t *, void *, size_t);
                        ^
  9 errors generated.
```
Full log: https://github.com/macports/macports-ports/actions/runs/3925116547/jobs/6709822109
https://github.com/macports/macports-ports/actions/runs/3925116547/jobs/6709822052

**Comment by chrberger** [comment]
fixed with https://github.com/chrberger/stringtoolbox/commit/eb1d24e848677de160c22dabb01b5d1272291d29

**Comment by barracuda156** [comment]
> fixed with [eb1d24e](https://github.com/chrberger/stringtoolbox/commit/eb1d24e848677de160c22dabb01b5d1272291d29)

Thank you!

**Comment by barracuda156** [comment]
@chrberger Ah, one more thing, sorry that I forgot: `-Wunused-but-set-parameter` and `-Wunused-but-set-variable` are not supported with Clang, and the build fails, I had to patch them out like:
```
post-patch {
     # Clang does not understand the following flags:
     if {[string match *clang* ${configure.compiler}]} {
         reinplace "s,-Wunused-but-set-parameter,," ${worksrcpath}/CMakeLists.txt
         reinplace "s,-Wunused-but-set-variable,," ${worksrcpath}/CMakeLists.txt
     }
 }
```
GCC is fine with them. If they are needed, perhaps should be conditional on `CMAKE_CXX_COMPILER_ID` not matching Clang.
